### ISSUE TITLE
Could you provide example code for AdaLoRA finetuning decoder-only model?

### ISSUE THREAD

**Issue by SpeeeedLee** [huggingface/peft#2262]
### Feature request

The current [example of AdaLoRA](https://github.com/huggingface/peft/blob/b2922565c4c4445706a87cf7b988c828b451fe61/examples/conditional_generation/peft_adalora_seq2seq.py) is on **facebook/bart-base**.  Since AdaLoRA requires hand-crafted calculations on loss, would it be possible to provide me some hints on how can this be done when it comes to decoder-only (e.g., Llama-Instruct) LM?

Specificially, I would like to mask out the loss calculation on the instruction part or system prompt, focusing only on the assistant response.

### Motivation

AdaLoRA requires hand-crafted calculations on loss, which becomes complex when desired to mask out some system/instructino tokens.

### Your contribution

N.A.

**Comment by d-kleine** [comment]
As there is response yet, I am trying to answer your question how I would approach this:

> Specificially, I would like to mask out the loss calculation on the instruction part or system prompt, focusing only on the assistant response.

My idea would be to work with the special tokens in the prompt. Depending on the model you are using (the prompt template varies across different model architectures), I would look for the `<|assistant|>` special token. As the special tokens `<|user|>` and `<|system|>` are typically before the <|assistant|> special token, my approach would be to write custom trainer class that masks the input starting with the `<|assistant|>` prompt and excludes everything before that special token.

**Comment by SpeeeedLee** [comment]
Thank you for your response. Yes, I have successfully written a custom code to set the labels of tokens before `<|assistant|>`  to `-100`. Then following the example code provided, a custom training loop is conducted. The results are reasonable.

**Comment by BenjaminBossan** [comment]
I'm glad to hear it worked, thanks for your suggestion @d-kleine.

If there are no further questions, feel free to close the issue @SpeeeedLee.
